### DISCUSSION TITLE
Sudo password prompt

### DISCUSSION THREAD

**Discussion by angelocordero** [sxyazi/yazi#435] [4. Ideas]
I have read [this](https://github.com/sxyazi/yazi/issues/307) and I know it's closed as not possible as 
> Yazi must always run as root

but would it be possible to have a builtin operation keyword similar that asks the user the password first, then pipe it to sudo and execute the command?

similar to the `shell` operation keyword but runs it as root instead, after asking for the password 

Sudo has a `-s` flag which allows it to 
> read password from standard input

although I understand that there probably are security risks involved in this approach

**Comment by sxyazi** [comment]
The copy and cut operations in Yazi directly use system APIs, occurring within the Yazi process itself rather than invoking external `cp` or `mv` commands. As far as I know, Linux permissions are specific to processes, meaning that elevation of privileges is not possible without creating a new process.

Invoking external `cp` or `mv` commands operates in a different mode, seemingly adopted by other terminal file managers. However, Yazi chose not to follow this approach for two reasons:

1. They cannot provide real-time progress updates.
2. They always run as single-threaded, whereas Yazi performs concurrent copying and moving of files, resulting in faster operations.

Nevertheless, I think that maybe we could perform tasks similar to `shell "sudo cp src/ dist/"`. For operations requiring actual elevation of privileges, users could be allowed to execute them in the form of an external plugin creating a new process.

This would require support from the plugin system, https://github.com/sxyazi/yazi/pull/401, and it should be soon. Would you like to try this out once it's ready?

**Reply by angelocordero** [reply]
Thanks. Waiting for it to be ready

**Comment by sxyazi** [comment]
There's a plugin to do this now https://github.com/TD-Sky/sudo.yazi

Also, Yazi now supports cross-process sync of yank states, meaning you can also copy and paste files between two Yazi processes, even if one runs with normal permissions and the other with sudo privileges, https://yazi-rs.github.io/docs/dds#session.lua
